src/Components/WebAssembly/testassets/ThreadingApp/Pages/Counter.razor

```diff
@@ -0,0 +1,98 @@
+﻿@page "/counter"
+@using System.Runtime.InteropServices
+@using System.Threading
+
+<h1>Counter</h1>
+
+<p>Current count: @currentCount</p>
+
+<button class="btn btn-primary" @onclick="IncrementCount">Click me</button>
+
+@code {
+    int currentCount = 0;
+    System.Threading.Timer timer;
+
+    void IncrementCount()
+    {
+        currentCount++;
+    }
+
+    protected override async Task OnInitializedAsync()
+    {
+        if (!OperatingSystem.IsBrowser())
+        {
+            return;
+        }
+
```

Comment:
```suggestion
        if (!OperatingSystem.IsBrowser())
        {
            return;
        }

```

Not sure we need to check this. Is there some reason for thinking this app might get run in another way?

Reply:
`OnInitializedAsync()` could also run on the server side, right ? And the rest of the code is not relevant on the server side.

Reply:
I thought this whole app was set up to run on WebAssembly, so I didn't think any part of it would ever run on the server.